pkg/kube/kclient/client_test.go

```diff
@@ -50,6 +51,35 @@ import (
 	"istio.io/istio/pkg/test/util/retry"
 )
 
+func TestSwappingClientIndex(t *testing.T) {
+	stop := test.NewStop(t)
+	c := kube.NewFakeClient()
+	wasm := kclient.NewDelayedInformer[controllers.Object](c, gvr.WasmPlugin, kubetypes.StandardInformer, kubetypes.Filter{})
+	c.RunAndWait(stop)
+	idx := kclient.CreateStringIndex(wasm, func(o controllers.Object) []string {
+		return []string{o.(*istioclient.WasmPlugin).Spec.ImagePullSecret}
+	})
+	assertIndex := func(k string, we ...controllers.Object) {
+		t.Helper()
+		assert.EventuallyEqual(t, func() []controllers.Object { return idx.Lookup(k) }, we, retry.Timeout(time.Second*5))
+	}
+	// To start, no response
+	assertIndex("secret1")
+	wt := clienttest.NewWriter[*istioclient.WasmPlugin](t, c)
+	wasm1 := &istioclient.WasmPlugin{
+		ObjectMeta: metav1.ObjectMeta{Name: "name", Namespace: "default"},
+		Spec:       ext.WasmPlugin{ImagePullSecret: "secret1"},
+	}
+	wt.Create(wasm1)
+	// Still no response since we haven't started yet
+	assertIndex("secret1")
+
+	// Make the CRD
+	clienttest.MakeCRD(t, c, gvr.WasmPlugin)
+	// Still no response since we haven't started yet
+	assertIndex("secret1", wasm1)
```

Comment:
should we start and assert we have an entry?

Reply:
That is what it does actually, just the comment is wrong

Reply:
Fixed, thanks